app/models/claim/matching_attribute_finder.rb

```diff
@@ -0,0 +1,31 @@
+class Claim
+  class MatchingAttributeFinder
+    ATTRIBUTES_TO_MATCH = %w[
+      teacher_reference_number
+      email_address
+      national_insurance_number
+      bank_account_number
+      bank_sort_code
+      building_society_roll_number
+    ].freeze
+
+    def initialize(source_claim, claims_to_compare = Claim.submitted)
+      @source_claim = source_claim
+      @claims_to_compare = claims_to_compare
+    end
+
+    def matching_claims
+      predicates = []
+      values = []
+      ATTRIBUTES_TO_MATCH.each do |attribute|
```

Comment:
I think it's OK to put the SQL in directly here (so we don't need the `predicates` and `values`) as the data is already stored in the database, so is already sanitised. So you could do something like:

```ruby
query = ATTRIBUTES_TO_MATCH.each do |attribute|
  value = @source_claim.read_attribute(attribute)
  next unless value

  "#{attribute} ='#{value}'"
end

@claims_to_compare.where.not(id: @source_claim.id).where(query)
```

Reply:
Hmm, I don't think so. What I mean is that the Ruby values need to be made suitable for insertion into an SQL query, that is, they need to be typecast (e.g. a `Date` needs to be turned into the DB's representation of a date) and stuff that shouldn't be in an SQL query (e.g. quotes) need to be escaped. We can't just drop the values in like that.

Reply:
Yes, what @lawrence-dxw said, we need to let ActiveRecord escape any user-provided values that we use in SQL queries, otherwise we open ourselves up to potential SQL-injection attacks.

Reply:
Ah, I guess. I didn't think about type casting here.